Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -0,0 +1,412 @@
+using Content.Server.Chat.Managers;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.Mind;
+using Content.Server.NPC.Systems;
+using Content.Server.Players;
+using Content.Server.Preferences.Managers;
+using Content.Server.Roles;
+using Content.Server.RoundEnd;
+using Content.Server.Station.Systems;
+using Content.Shared.Humanoid;
+using Content.Shared.Mobs;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Preferences;
+using Content.Shared.Revolutionary.Components;
+using Content.Shared.Roles;
+using Robust.Server.Player;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using System.Linq;
+using Content.Shared.Stunnable;
+using Content.Server.Chat.Systems;
+using Content.Server.Shuttles.Components;
+using Robust.Shared.Timing;
+using Content.Server.Popups;
+using Content.Server.Revolutionary.Components;
+using Content.Server.Mindshield;
+using Content.Server.Shuttles.Systems;
+using Content.Server.Objectives.Interfaces;
+using Content.Shared.IdentityManagement;
+
+namespace Content.Server.GameTicking.Rules;
+
+/// <summary>
+/// Where all the main stuff for Revolutionaries happens (Assigning Head Revs, Command on station, and checking for game the game to end.)
+/// </summary>
+
+public sealed class RevolutionaryRuleSystem : GameRuleSystem<RevolutionaryRuleComponent>
+{
```

Comment:
This system needs a lot of cleaning up but I'll go through it in more detail later.